### DISCUSSION TITLE
How to un-delete old commits

### DISCUSSION THREAD

**Discussion by FrankiePolak** [community/community#21676] [New to GitHub]
<p>I was having some problems with GitHub, and in messing around trying to fix them, I realized that I’d deleted some of my work. I followed <a href="https://www.christianengvall.se/git-reset-origin-master-to-commit/" rel="nofollow noopener">these instructions</a> to reset to the commit before I started messing around, but now all of my previous commits have disappeared. Is there a way to get them back?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>If you want to restore your GitHub repository to an older state, you can contact private support at <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a>. I’m not sure what “previous commits” you are saying disappeared, though. If you could be more specific about what you did and what exactly disappeared, there <em>might</em> be other options available.</p>
<p>Let us know if you have more questions.</p>

**Comment by FrankiePolak** [comment]
<p>Thanks for the response!</p>
<p>I meant that I used to have 57 commits on GitHub, and I accidentally deleted all but the most recent one. My problem is solved now, though.</p>
<p>If anyone else has this issue, you just need to message GitHub support.</p>
